### DISCUSSION TITLE
POSTing a job to API

### DISCUSSION THREAD

**Discussion by BreMJ** [NREL/REopt-API-Analysis#86] [8 - API: Application Programming Interface]
Hello,

I am attempting to run the API locally. I am following the instructions for running the API locally using docker. I have gotten to step 4 where I am asked to make a POST and GET. When I attempt this, I do not receive a run_uuid for the GET URL. The response I receive is "POST" with no additional information. I think I might be missing a step when establishing my inputs and POST but I am not sure.

**Comment by Bill-Becker** [comment]
Can you please confirm that you are using an open source solve like CBC and that you used that version of the docker-compose file? Also, can you tell me what URL you are using to POST to your locally-hosted API? Are you using the REopt-API-Analysis repository to POST/GET to your locally hosted API, and if so are you using the Docker setup for the Jupyter Lab/Notbook?

**Reply by BreMJ** [reply]
Yes, I can confirm that I am using CBC solver and the version of the docker-compose file. The URL I attempted to POST to is https://developer.nrel.gov/api/reopt. I am using the REopt-API-Analysis repository to POST/GET, but I have not been able to access the Jupyter Lab/Notebook. I did not receive an URL to access Jupyter Lab in my browser upon running the docker-compose build up.

**Reply by Bill-Becker** [reply]
Just to clarify, the Jupyter Lab URL is available when you `docker compose up` from the **REopt-API-Analysis** repository, not from `docker compose up` of the `REopt_API` code which is for hosting the API locally. Then, you should get a URL/link in the terminal for your localhost; something like http://127.0.0.1:8888/lab. From there, if you open up the Jupyter Notebook (https://github.com/NREL/REopt-API-Analysis/blob/master/single_scenario_example.ipynb), you will see instructions as you go down for calling your locally-hosted API.

**Reply by BreMJ** [reply]
Ok, I understand. Thank you for the clarification.
